stream_alert/shared/lookup_tables/driver_dynamodb.py

```diff
@@ -0,0 +1,194 @@
+        not possible to have a non-string type for either of these.
+    """
+    def __init__(self, configuration):
+        # Example configuration:
+        # {
+        #     "driver": "dynamodb",
+        #     "table": "some_table_name",
+        #     "partition_key": "MyPartitionKey",
+        #     "sort_key": "MySortKey",
+        #     "value_key": "MyValueKey",
+        #     "cache_refresh_minutes": 2,
+        #     "cache_maximum_key_count": 10,
+        #     "consistent_read": false,
+        #     "key_delimiter": ":"
+        # }
+
+        super(DynamoDBDriver, self).__init__(configuration)
+
+        self._dynamo_db_table = configuration['table']
+        self._dynamo_db_partition_key = configuration['partition_key']
+        self._dynamo_db_value_key = configuration['value_key']
+        self._dynamo_db_sort_key = configuration.get('sort_key', False)
+        self._dynamo_consistent_read = configuration.get('consistent_read', True)
+
+        self._dynamo_data = {}
+        self._dynamo_load_times = {}
+
+        self._cache_maximum_key_count = configuration.get('cache_maximum_key_count', 1000)
+        self._cache_refresh_minutes = configuration.get('cache_refresh_minutes', 3)
+
+        self._key_delimiter = configuration.get('key_delimiter', ':')
+
+        self._table = None
+
+    @property
+    def driver_type(self):
+        return self.TYPE_DYNAMODB
+
+    @property
+    def id(self):
+        return '{}:{}'.format(self.driver_type, self._dynamo_db_table)
+
+    def initialize(self):
+        # Setup DynamoDb client
+        LOGGER.info('LookupTable (%s): Running initialization routine', self.id)
+
+        try:
+            boto_config = boto_helpers.default_config(timeout=10)
+            resource = boto3.resource('dynamodb', config=boto_config)
+            self._table = resource.Table(self._dynamo_db_table)
+            _ = self._table.table_arn  # This is only here to blow up on invalid tables
+        except ClientError as err:
+            message = (
+                'LookupTable ({}): Encountered error while connecting with DynamoDB: \'{}\''
+            ).format(self.id, err.response['Error']['Message'])
+            LOGGER.error(message)
+            raise LookupTablesInitializationError(message)
+
+    def commit(self):
+        # FIXME (derek.wang)
+        pass
+
+    def get(self, key, default=None):
+        self._reload_if_necessary(key)
+
+        return self._dynamo_data.get(key, default)
+
+    def set(self, key, value):
+        # FIXME (derek.wang)
+        pass
+
+    def _reload_if_necessary(self, key):
+        """
+        Uses the "cache_refresh_minutes" option to determine whether or not the current LookupTable
+        should be re-fetched from S3.
+
+        If it needs a reload, this method will appropriately call reload.
+        """
+        last_load_time = self._dynamo_load_times.get(key, 0)
```

Comment:
neato

Reply:
I refactored this significantly later. There's a new class called `DriverCache`